x/config/backoff.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package config
+
+import (
+	"time"
+
+	"go.uber.org/yarpc/internal/backoff"
+)
+
+// Backoff specifies a backoff strategy, particularly for retries.
+// The only supported strategy at time of writing is "exponential" with full
+// jitter. This structure may be extended in the future to support registering
+// alternate backoff strategies.
+//
+//  exponential:
+//    min: 100ms
+//    base: 100ms
+//    max: 30s
+type Backoff struct {
+	Exponential ExponentialBackoff `config:"exponential"`
+}
```

Comment:
Should we make the fields pointers so that we can tell which of the strategies is enabled? The zero value of an exponential strategy is clearly invalid, but I'm not sure that'll be true of all future strategy configurations.